waspc/data/Generator/templates/react-app/src/queryClient.js

```diff
@@ -1,3 +1,16 @@
 import { QueryClient } from 'react-query'
 
-export const queryClient = new QueryClient()
+let resolveQueryClientInitialized, isQueryClientSetUp
+export const queryClientInitialized = new Promise(resolve => {
```

Comment:
I found this name a bit confusing, with the "initialized" part.
Could we instead just call it `queryClient`? The fact that there is no `queryClient` until we call `setupQueryClient` is a separate thing, it doesn't necessarily need to be reflected in the name of `queryClient`.

Reply:
It creates a bit of a name clash.

It makes sense for the user code to use the name `queryClient` for a true instance of the `QueryClient`.
```javascript
// This wouldn't work.
// User code would either have to come up with an alternative name for the awaited instance
// or use an import alias
const queryClient = await queryClient;
```

Reply:
Good point @sodic , ok let's go with `queryClientInitialized`! Maybe `initializedQueryClient` is nicer? But ok both are fine.

Reply:
Aha but that then looks very similar to `initializeQueryClient`, ok makes sense, better keep it `queryClientInitialized`.